homeassistant/components/climate/generic_thermostat.py

```diff
@@ -198,24 +198,32 @@ def _control_heating(self):
                 return
 
         if self.ac_mode:
-            too_hot = self._cur_temp - self._target_temp > self._tolerance
+
             is_cooling = self._is_device_active
-            if too_hot and not is_cooling:
-                _LOGGER.info('Turning on AC %s', self.heater_entity_id)
-                switch.turn_on(self.hass, self.heater_entity_id)
-            elif not too_hot and is_cooling:
-                _LOGGER.info('Turning off AC %s', self.heater_entity_id)
-                switch.turn_off(self.hass, self.heater_entity_id)
+            if not is_cooling:
```

Comment:
This would make more sense if we switch the if to be `if is_cooling:`, then just `else`